app/controllers/api/v1/statuses/histories_controller.rb

```diff
@@ -8,11 +8,16 @@ class Api::V1::Statuses::HistoriesController < Api::BaseController
 
   def show
     cache_if_unauthenticated!
-    render json: @status.edits.includes(:account, status: [:account]), each_serializer: REST::StatusEditSerializer
+    render json: status_edits, each_serializer: REST::StatusEditSerializer
   end
 
   private
 
+  def status_edits
+    @status_edits ||= @status.edits.includes(:account, status: [:account]).to_a
+    @status_edits.presence || [@status.build_snapshot(at_time: @status.edited_at || @status.created_at)]
```

Comment:
This is not cached if `@status_edits` is empty, should it be?

Reply:
I don't think this needs caching at all, I'll simplify the code.